src/pushsource/_impl/backend/koji_source.py

```diff
@@ -283,18 +284,38 @@ def _push_items_from_rpm_meta(self, rpm, meta):
 
         candidate_paths = []
 
-        # If signing keys requested, try them in order of preference
-        for key in self._signing_key:
-            if key:
-                key = key.lower()
-                candidate = os.path.join(build_path, self._pathinfo.signed(meta, key))
-            else:
-                candidate = unsigned_path
-            candidate_paths.append(candidate)
-            if os.path.exists(candidate):
-                rpm_path = candidate
-                rpm_signing_key = key
+        timeout = int(os.getenv("PUSHSOURCE_SRC_POLL_TIMEOUT") or "0")
+        poll_rate = int(os.getenv("PUSHSOURCE_SRC_POLL_RATE") or "30")
+        max_attempts = timeout // poll_rate
+
+        for i in range(max_attempts + 1):
+            # We only want to poll if signing key is requested
+            if not self._signing_key:
                 break
+            # If signing keys requested, try them in order of preference
+            for key in self._signing_key:
+                if key:
+                    key = key.lower()
+                    candidate = os.path.join(
+                        build_path, self._pathinfo.signed(meta, key)
+                    )
+                else:
+                    candidate = unsigned_path
+                if candidate not in candidate_paths:
+                    candidate_paths.append(candidate)
```

Comment:
Since this is in a loop and candidate_paths is a list, aren't we putting duplicates in the list now?  It'll at least mess up the LOG.info below.

Reply:
The code is changed now so this is no longer a concern, but I did add a condition `if candidate not in candidate_paths:`